SecretSanta/src/SecretSanta.Business/EntityService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using AutoMapper;
+using SecretSanta.Data;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace SecretSanta.Business
+{
+    public abstract class EntityService<TEntity> : IEntityService<TEntity>
+        where TEntity : EntityBase
+    {
+        protected ApplicationDbContext ApplicationDbContext { get; }
+        protected IMapper Mapper { get; }
+
+        public EntityService(ApplicationDbContext applicationDbContext, IMapper mapper)
+        {
+            ApplicationDbContext = applicationDbContext ?? throw new ArgumentNullException(nameof(applicationDbContext));
+            Mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
+        }
+        virtual public Task<bool> DeleteAsync(int id)
+        {
+            throw new NotImplementedException();
```

Comment:
```suggestion
            throw new NotImplementedException();
```
Even though it is not actively used in this assignment- I believe this should still be implemented & tested as it will likely be used for future implementations of this project.